File: Data Structures & Algorithms/daily-temperatures/submission-1.js
class Solution {
    /**
     * @param {number[]} temperatures
     * @return {number[]}
     */
    dailyTemperatures(t) {
        let result = [];
        for ( let i = 0; i< t.length;i++){
            let curr = t[i];
            let j = i+1;
            while (curr >= t[j] && j < t.length){
                j++;
            }
            if (curr < t[j]) result.push(j-i)
            else result.push(0);
        }
        return result;
    }
}

File: Data Structures & Algorithms/permutations/submission-1.py
class Solution:
    def permute(self, nums: List[int]) -> List[List[int]]:
        result = []
        path = []

        def dfs():
            if len(path) == len(nums):
                result.append(path[:])
                return
            
            for i in range(len(nums)):
                curr = nums[i]
                if curr in path:
                    continue
                path.append(curr)
                dfs()
                path.pop()
        dfs()
        
        return result
